core/pom.xml

```diff
@@ -42,6 +42,10 @@
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-api</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.google.code.findbugs</groupId>
+            <artifactId>jsr305</artifactId>
+        </dependency>
```

Comment:
Why do you add this dependency? It doesn't seem to be used anywhere.

Reply:
This dependency provides the `javax.annotation.Nonnull` and `javax.annotation.Nullable` annotations from JSR-305.

Alternatively we could use some other vendor-specific annotations with similar purpose, for example https://github.com/JetBrains/java-annotations